src/pybind/mgr/volumes/fs/async_job.py

```diff
@@ -35,24 +37,39 @@ def run(self):
         while retries < JobThread.MAX_RETRIES_ON_EXCEPTION:
             vol_job = None
             try:
+                if not self.async_job.run_event.is_set():
+                    log.info('will wait for run_event to bet set. postponing '
+                             'getting jobs until then')
+                    self.async_job.run_event.wait()
+                    log.info('run_event has been set, waiting is complete. '
+                             'proceeding to get jobs now')
+
                 # fetch next job to execute
                 with lock_timeout_log(self.async_job.lock):
                     while True:
                         if self.should_reconfigure_num_threads():
                             log.info("thread [{0}] terminating due to reconfigure".format(thread_name))
                             self.async_job.threads.remove(self)
                             return
+
                         timo = self.async_job.wakeup_timeout
                         if timo is not None:
                             volnames = list_volumes(self.vc.mgr)
                             missing = set(volnames) - set(self.async_job.q)
                             for m in missing:
                                 self.async_job.jobs[m] = []
                                 self.async_job.q.append(m)
+
+                        if not self.async_job.run_event.is_set():
+                            break
+
                         vol_job = self.async_job.get_job()
                         if vol_job:
                             break
                         self.async_job.cv.wait(timeout=timo)
+
+                    if not self.async_job.run_event.is_set():
```

Comment:
and this too?

Reply:
When user pauses before threads find the job, the `break` we discussed in above previous comment will take care of pausing. If user pauses while job is being executed, `should_cancel` in `purge()` of `operation/trash.py` will take care of pausing.

But what if we user pauses while thread is between getting the job and executing it? `continue` under the `if` takes care of this case.

Reply:
Marking this comment resolved since it was discussed and changes have moved forward since.